jax/_src/lax/slicing.py

```diff
@@ -1287,10 +1295,17 @@ def _gather_batching_rule(batched_args, batch_dims, *, dimension_numbers,
         offset_dims=offset_dims,
         collapsed_slice_dims=collapsed_slice_dims,
         start_index_map=start_index_map)
-    return gather(operand, indices, dimension_numbers=dnums,
-                  slice_sizes=slice_sizes, unique_indices=unique_indices,
+    # TODO(reviewer): This should be correct for `operand_bdim` being a
+    # `RaggedAxis` as long as we are not gathering from any ragged axis, and as
+    # long as we have no collapsed_slice_dims.  (The latter could require
+    # adjusting the `ragged_axes` field of `operand_bdim`).  Should we put a
+    # check to confirm?
```

Comment:
Let's definitely put a check! Any reason not to? (We can raise NotImplemented in unhandled cases iiuc).

Reply:
Shape prediction improved and check added.